### ISSUE TITLE
Move main to  maven.compiler.release -> 17 

### ISSUE THREAD

**Issue by wmedvede** [apache/incubator-kie-kogito-runtimes#3343]
### Describe the bug

Guys, after moving main to quarkus 3.2.9.Final, where it's supposed that we are now in jdk-17, is there any reason for having this in kogito-runtimes?

```
<maven.compiler.release>11</maven.compiler.release>
<maven.compiler.source>11</maven.compiler.source>
<maven.compiler.target>11</maven.compiler.target>
```

see:
https://github.com/apache/incubator-kie-kogito-runtimes/blob/1c560394c2bf75c53388afd42892acbb05d0698b/kogito-build/kogito-build-no-bom-parent/pom.xml#L81

I can see some tests failing because of this, for example kogito-runtimes/quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-image-integration-test, when quarkus-container-image-jib picks a base image to build images, it'll pick a ubi8/openjdk-11-runtime  as base image.

But we have generated code, for example, org/kie/api/definition/process/Process produced by drools, that is in java-17
in summary, these generated image will fail complaining about:

Caused by: java.lang.UnsupportedClassVersionError: org/kie/api/definition/process/Process has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0

note that drools now has:

```
<maven.compiler.release>17</maven.compiler.release>
<maven.compiler.source>${maven.compiler.release}</maven.compiler.source>
<maven.compiler.target>${maven.compiler.release}</maven.compiler.target>
```

https://github.com/apache/incubator-kie-drools/blob/0c0b3e5d6e1a02ed46f1c31de359ac876997834b/build-parent/pom.xml#L38

**Comment by mariofusco** [comment]
@wmedvede that's for sure a leftover, if you have time please send a PR fixing it.

**Comment by wmedvede** [comment]
This was fixed by:

https://github.com/apache/incubator-kie-kogito-runtimes/pull/3344
https://github.com/apache/incubator-kie-kogito-apps/pull/1944
